libenv/unix_iface.c

```diff
@@ -752,4 +755,444 @@ static int aix_get_mac_addr(const char *device_name, uint8_t mac[6])
+        BufferPrintf(formatter, "ipv4_gw_%s", JsonObjectGetAsString(route, "gateway"));
+        EvalContextClassPutHard(ctx, BufferData(formatter), "inventory,networking,/proc,source=agent,attribute_name=none,procfs");
+        BufferDestroy(formatter);
+    }
+# endif
+    return NULL;
+}
+
+static ProcPostProcessFn NetworkingIPv6RoutesPostProcessInfo(ARG_UNUSED void *passed_ctx, void *json)
+{
+# if defined (__linux__)
+    JsonElement *route = json;
+
+    JsonRewriteParsedIPAddress(route, "raw_dest", "dest", false);
+    JsonRewriteParsedIPAddress(route, "raw_next_hop", "next_hop", false);
+    JsonRewriteParsedIPAddress(route, "raw_source", "dest", false);
+
+    JsonExtractParsedNumber(route, "raw_metric", "metric", true, false);
+    JsonExtractParsedNumber(route, "refcnt", "refcnt", false, false);
+    JsonExtractParsedNumber(route, "use", "use", false, false);
+
+    JsonElement *decoded_flags = JsonArrayCreate(3);
+    long num_flags = JsonExtractParsedNumber(route, "raw_flags", NULL, true, false);
+
+    bool is_up = (num_flags & RTF_UP);
+    bool is_gw = (num_flags & RTF_GATEWAY);
+    bool is_host = (num_flags & RTF_HOST);
+
+    const char* gw_type = is_gw ? "gateway":"local";
+
+    // These flags are always included on Linux in platform.h
+    JsonArrayAppendString(decoded_flags, is_up ? "up":"down");
+    JsonArrayAppendString(decoded_flags, is_host ? "host":"net");
+    JsonArrayAppendString(decoded_flags, gw_type);
+    JsonObjectAppendElement(route, "flags", decoded_flags);
+
+    // TODO: figure out if we can grab any default gateway info here
+    // like we do with IPv4 routes
+
+# endif
+    return NULL;
+}
+
+static ProcPostProcessFn NetworkingIPv6AddressesPostProcessInfo(ARG_UNUSED void *passed_ctx, void *json)
+{
+    JsonElement *entry = json;
+
+    JsonRewriteParsedIPAddress(entry, "raw_address", "address", false);
+
+    JsonExtractParsedNumber(entry, "raw_device_number", "device_number", true, false);
+    JsonExtractParsedNumber(entry, "raw_prefix_length", "prefix_length", true, false);
+    JsonExtractParsedNumber(entry, "raw_scope", "scope", true, false);
+
+    return NULL;
+}
+
+/*******************************************************************/
+
+static const char* GetPortStateString(int state)
+{
+# if defined (__linux__)
+    switch (state)
+    {
+    case TCP_ESTABLISHED: return "ESTABLISHED";
+    case TCP_SYN_SENT:    return "SYN_SENT";
+    case TCP_SYN_RECV:    return "SYN_RECV";
+    case TCP_FIN_WAIT1:   return "FIN_WAIT1";
+    case TCP_FIN_WAIT2:   return "FIN_WAIT2";
+    case TCP_TIME_WAIT:   return "TIME_WAIT";
+    case TCP_CLOSE:       return "CLOSE";
+    case TCP_CLOSE_WAIT:  return "CLOSE_WAIT";
+    case TCP_LAST_ACK:    return "LAST_ACK";
+    case TCP_LISTEN:      return "LISTEN";
+    case TCP_CLOSING:     return "CLOSING";
+    }
+
+# endif
+    return "UNKNOWN";
+}
+
+static ProcPostProcessFn NetworkingPortsPostProcessInfo(ARG_UNUSED void *passed_ctx, void *json)
```

Comment:
`NetworkingPortsPostProcessInfo()` and `GetPortStateString()` should be moved to #2495. I can't see that they are needed in this pull request. This is important if we need to revert anything later.